### ISSUE TITLE
ZED: Opt-Out flooding syslog with Invoking / Finished messages

### ISSUE THREAD

**Issue by spacerunner5** [openzfs/zfs#15862]
### Describe the feature would like to see added to OpenZFS

Add the option to disable logging of [zed] Invoking... and [zed] Finished.. messages.

### How will this feature improve OpenZFS?

Cleaner log files. 

### Current state
Log file example:
```
2024-02-06T16:17:47.715417+01:00 PVE1 zed[20251]: Finished "all-syslog.sh" eid=18 pid=20325 time=0.001621s exit=0
2024-02-06T16:17:47.715471+01:00 PVE1 zed[20251]: Finished "history_event-zfs-list-cacher.sh" eid=18 pid=20326 time=0.001366s exit=0
2024-02-06T16:17:47.715920+01:00 PVE1 zed[20251]: Invoking "all-syslog.sh" eid=19 pid=20331
2024-02-06T16:17:47.716140+01:00 PVE1 zed[20251]: Invoking "history_event-zfs-list-cacher.sh" eid=19 pid=20333
2024-02-06T16:17:47.717413+01:00 PVE1 zed[20251]: Finished "history_event-zfs-list-cacher.sh" eid=19 pid=20333 time=0.001234s exit=0
2024-02-06T16:17:47.717848+01:00 PVE1 zed[20251]: Invoking "all-syslog.sh" eid=20 pid=20340
2024-02-06T16:17:47.717948+01:00 PVE1 zed[20251]: Finished "all-syslog.sh" eid=19 pid=20331 time=0.001770s exit=0
2024-02-06T16:17:47.718068+01:00 PVE1 zed[20251]: Invoking "history_event-zfs-list-cacher.sh" eid=20 pid=20341
2024-02-06T16:17:47.719231+01:00 PVE1 zed[20251]: Finished "all-syslog.sh" eid=20 pid=20340 time=0.001338s exit=0
2024-02-06T16:17:47.719269+01:00 PVE1 zed[20251]: Finished "history_event-zfs-list-cacher.sh" eid=20 pid=20341 time=0.001179s exit=0
2024-02-06T16:17:47.719821+01:00 PVE1 zed[20251]: Invoking "all-syslog.sh" eid=21 pid=20344
2024-02-06T16:17:47.720087+01:00 PVE1 zed[20251]: Invoking "history_event-zfs-list-cacher.sh" eid=21 pid=20345
2024-02-06T16:17:47.721408+01:00 PVE1 zed[20251]: Finished "history_event-zfs-list-cacher.sh" eid=21 pid=20345 time=0.001280s exit=0
2024-02-06T16:17:47.721722+01:00 PVE1 zed[20251]: Invoking "all-syslog.sh" eid=22 pid=20350
2024-02-06T16:17:47.721936+01:00 PVE1 zed[20251]: Invoking "history_event-zfs-list-cacher.sh" eid=22 pid=20353
2024-02-06T16:17:47.721999+01:00 PVE1 zed[20251]: Finished "all-syslog.sh" eid=21 pid=20344 time=0.002042s exit=0
2024-02-06T16:17:47.723106+01:00 PVE1 zed[20251]: Finished "history_event-zfs-list-cacher.sh" eid=22 pid=20353 time=0.001178s exit=0
2024-02-06T16:17:47.723445+01:00 PVE1 zed[20251]: Invoking "all-syslog.sh" eid=23 pid=20361
2024-02-06T16:17:47.723675+01:00 PVE1 zed[20251]: Invoking "history_event-zfs-list-cacher.sh" eid=23 pid=20362
2024-02-06T16:17:47.723703+01:00 PVE1 zed[20251]: Finished "all-syslog.sh" eid=22 pid=20350 time=0.001986s exit=0
2024-02-06T16:17:47.724536+01:00 PVE1 zed[20251]: Finished "history_event-zfs-list-cacher.sh" eid=23 pid=20362 time=0.000892s exit=0
[AND SO ON]
```

### Additional context

ZED must be running. The problem especially grows logs when running frequent cron jobs, such as for automatic snapshot creation / pruning / rotating. Every time a zfs command is issued, ZED logs those messages for all involved ZEDLETs. This can easily lead to **thousands** of messages per day.

ZEDLETs log output can be filtered quite nice by modifying their scripts. This can even be user extended by modifying @sysconfdir@/zfs/zed.d/zed.rc.

There is however no (documented?) control of the log messages mentioned above, since their output is **hardcoded** in https://github.com/openzfs/zfs/blob/6dccdf501ea47bb8a45f00e4904d26efcb917ad4/cmd/zed/zed_exec.c#L188
and
https://github.com/openzfs/zfs/blob/6dccdf501ea47bb8a45f00e4904d26efcb917ad4/cmd/zed/zed_exec.c#L240

That's why setting for example setting zed.rc::ZED_SYSLOG_SUBCLASS_EXCLUDE is no solution here: The ZEDLETs invoked in the SUBCLASS this get loaded. Although they exit quickly due to calling zed-functions.sh::zed_exit_if_ignoring_this_event(), they get loaded, invoked and logged as above.

ZED should have an (config) option to bypass those messages during normal operation. They are purely informational. Dumping them  is great for debugging and they should be fired on case of an error.

Reference: #8755

Side note: To make it a bit more confusing, it should be noted that two zed_log_msg() functions exist - one in 
https://github.com/openzfs/zfs/blob/6dccdf501ea47bb8a45f00e4904d26efcb917ad4/cmd/zed/zed_log.c#L231
and the scripted version in
https://github.com/openzfs/zfs/blob/6dccdf501ea47bb8a45f00e4904d26efcb917ad4/cmd/zed/zed.d/zed-functions.sh#L58

### workarround in Linux (informational)

The only way to get rid of the flood is filtering in the syslog itself, prior to logging. But that is of course nasty, since it may prevent ZEDLET Invoking errors of being logged.

Debian example with journald: Place an service config override in 

```
# file /etc/systemd/system/zfs-zed.service.d/override.conf
[Service]
LogFilterPatterns=~Invoking
LogFilterPatterns=~Finished
```
or edit directly using
```
~ # systemctl edit zfs-zed.service
```

**Comment by rincebrain** [comment]
You could also just...unlink the zedlets you don't want to run in `/etc/zfs/zed.d`.

**Comment by spacerunner5** [comment]
> You could also just...unlink the zedlets you don't want to run in `/etc/zfs/zed.d`.

…but that does not fix the problem for ZEDLETs **actually needed**. The log output above is just a example for being generic.

Example: I developed a custom ZEDLET invoked by _history_event_ for some simple snapshot manipulation. Same issue of course.

**Comment by rincebrain** [comment]
That's true, I misunderstood your lament as being about there being no way to disable zedlets, not the "start/stop" prints.

You could pretty readily patch in a config option to remove that output, if it's useful to you, and I doubt there'd be any complaints about it being merged.

**Comment by spacerunner5** [comment]
> You could pretty readily patch in a config option to remove that output, if it's useful to you, and I doubt there'd be any complaints about it being merged.

Unfortunately, I currently have no resources to C development. And unless being able to deeply test, I don’t want to mess with code. That‘s why I ask here for help. See next post for an idea how to solve this.

**Comment by JBlond** [comment]
Opt out via zed.rc

in /etc/zfs/zed.d/zed.rc

```ini
ZED_SYSLOG_SUBCLASS_EXCLUDE="history_event"
```
